devtools/utils.py

```diff
@@ -1,11 +1,100 @@
+import os
 import sys
 
 __all__ = ('isatty',)
 
+MYPY = False
+if MYPY:
+    from typing import Optional
+
 
 def isatty(stream=None):
     stream = stream or sys.stdout
     try:
         return stream.isatty()
     except Exception:
         return False
+
+
+def env_true(var_name: str, alt: 'Optional[bool]' = None) -> 'Optional[bool]':
+    env = os.getenv(var_name, None)
+    if env:
+        return env.upper() in {'1', 'TRUE'}
+    else:
+        return alt
+
+
+def env_bool(value: 'Optional[bool]', env_name: str, env_default: 'Optional[bool]') -> 'Optional[bool]':
+    if value is None:
+        return env_true(env_name, env_default)
+    else:
+        return value
+
+
+def activate_win_color() -> bool:  # pragma: no cover
+    """Activate ANSI support on windows consoles.
+
+    As of Windows 10, the windows conolse got some support for ANSI escape
+    sequences. Unfortunately it has to be enabled first using `SetConsoleMode`.
+    See: https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences
+
+    Code snippet source: https://bugs.python.org/msg291732
+    """
+    import os
+    import msvcrt
+    import ctypes
+
+    from ctypes import wintypes
+
+    def _check_bool(result, func, args):
+        if not result:
+            raise ctypes.WinError(ctypes.get_last_error())
+        return args
+
+    ERROR_INVALID_PARAMETER = 0x0057
+    ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004
+
+    LPDWORD = ctypes.POINTER(wintypes.DWORD)
+
+    kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+    kernel32.GetConsoleMode.errcheck = _check_bool
+    kernel32.GetConsoleMode.argtypes = (wintypes.HANDLE, LPDWORD)
+    kernel32.SetConsoleMode.errcheck = _check_bool
+    kernel32.SetConsoleMode.argtypes = (wintypes.HANDLE, wintypes.DWORD)
+
+    def _set_conout_mode(new_mode, mask=0xFFFFFFFF):
```

Comment:
is there a reason this is a function, not inline below?

Reply:
I think for readability reasons. Sure you can nest those two try/except blocks, but IMO this is more readable.

If you want me to I can refactor it.

Reply:
I guess my feeling is that the extra function is less readable and a little more confusion - normally nested functions are only used when:
1. you need to avoid duplicated code
2. you need closure to access variables used within the function

I read this and immediately ask "where else is it being called, oh why not?", but this is fine. I'll merge once tests pass.